packages/@aws-cdk/aws-iot-actions/lib/iotevents-put-message-action.ts

```diff
@@ -0,0 +1,72 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import * as iotevents from '@aws-cdk/aws-iotevents';
+import { CommonActionProps } from './common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for the IoT Events.
+ */
+export interface IotEventsPutMessageActionProps extends CommonActionProps {
+  /**
+   * Whether to process the event actions as a batch.
+   *
+   * When batchMode is true, you can't specify a messageId.
+   *
+   * When batchMode is true and the rule SQL statement evaluates to an Array,
+   * each Array element is treated as a separate message when Events by calling BatchPutMessage.
+   * The resulting array can't have more than 10 messages.
```

Comment:
If the array does have more than 10 messages how/when does this fail? What's the user experience here? We should probably have some enforcement here, if possible.

Reply:
Not explained in [the documentation](https://docs.aws.amazon.com/iot/latest/developerguide/iotevents-rule-action.html), I've experimented it with the below command:

```
aws iot-data publish --topic device/mydevice/data --qos 1 --payload (echo '[{"payload":{"deviceId":"001"}},{"payload":{"deviceId":"002"}},{"payload":{"deviceId":"003"}},{"payload":{"deviceId":"004"}},{"payload":{"deviceId":"005"}},{"payload":{"deviceId":"006"}},{"payload":{"deviceId":"007"}},{"payload":{"deviceId":"008"}},{"payload":{"deviceId":"009"}},{"payload":{"deviceId":"010"}},{"payload":{"deviceId":"011"}}]' | base64) --region us-east-1
```

and get below error message:

```
Failed to send message to Iot Events. The payload containing 11 messages cannot have more than 10 messages when batchMode=true.. 
```

I thing it is not possible to restrict it, because it depends to published payloads that is out of CDK.
WDYT?